### PR TITLE
test: 🧪 Added utilities test cases

### PR DESCRIPTION
**Pull request by Sahil-Simform**
# Description
- Added test cases to check utilities values

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [ ] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/SimformSolutionsPvtLtd/flutter_showcaseview/blob/master/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: test/utils_test.dart [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+      );
+      expect(
+        Constants.defaultScrollDuration,
+        const Duration(milliseconds: 300),
+      );
+      expect(
+        Constants.defaultProgressIndicator,
+        isA<ShowcaseCircularProgressIndicator>(),
+      );
+      expect(Constants.defaultTargetShapeBorder, isA<RoundedRectangleBorder>());
```
Comment (aditya-css) [inline test/utils_test.dart:27]: Please check the `borderRadius` as well.

File: test/utils_test.dart [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+    });
+
+    test(
+        'Constants progress indicator is instance of ShowcaseCircularProgressIndicator',
```
Comment (aditya-css) [inline test/utils_test.dart:31]: Already covered.

File: test/utils_test.dart [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+        () {
+      const progressIndicator = Constants.defaultProgressIndicator;
+      expect(progressIndicator, isA<ShowcaseCircularProgressIndicator>());
+    });
+
+    test('Constants target shape border is instance of RoundedRectangleBorder',
```
Comment (aditya-css) [inline test/utils_test.dart:37]: Already covered.

File: test/utils_test.dart [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+    });
+
+    test('TooltipPosition enum string values', () {
+      expect(TooltipPosition.top.toString(), 'TooltipPosition.top');
+      expect(TooltipPosition.bottom.toString(), 'TooltipPosition.bottom');
+      expect(TooltipPosition.left.toString(), 'TooltipPosition.left');
+      expect(TooltipPosition.right.toString(), 'TooltipPosition.right');
+    });
+
+    test('TooltipPosition enum equality', () {
```
Comment (aditya-css) [inline test/utils_test.dart:60]: `==` operator can't be overridden for an enum so this check seems unnecessary.

File: test/utils_test.dart [hunk 5]
```diff
@@ -0,0 +1,481 @@
...
+    });
+
+    test('Constants target shape border is instance of RoundedRectangleBorder',
+        () {
+      const targetShapeBorder = Constants.defaultTargetShapeBorder;
+      expect(targetShapeBorder, isA<RoundedRectangleBorder>());
+    });
+  });
+
+  group('Enum Tests', () {
```
Comment (aditya-css) [inline test/utils_test.dart:44]: You can test for enum's properties' default values too.

File: test/utils_test.dart [hunk 6]
```diff
@@ -0,0 +1,481 @@
...
+    });
+
+    test('TooltipActionConfig with default values', () {
+      const config = TooltipActionConfig();
+
+      expect(config.alignment, MainAxisAlignment.spaceBetween);
+      expect(config.actionGap, 5.0);
+    });
+
+    test('TooltipDefaultActionType enum values', () {
```
Comment (aditya-css) [inline test/utils_test.dart:101]: Placed under incorrect group.

File: test/utils_test.dart [hunk 7]
```diff
@@ -0,0 +1,481 @@
...
+    });
+
+    test('TooltipPosition enum equality', () {
+      expect(TooltipPosition.top == TooltipPosition.top, true);
+      expect(TooltipPosition.top == TooltipPosition.bottom, false);
+      expect(TooltipPosition.bottom == TooltipPosition.left, false);
+    });
+  });
+
+  group('Model Tests', () {
```
Comment (aditya-css) [inline test/utils_test.dart:67]: Testing default values are okay, but testing custom provided values doesn't seem to bring any benefits.

File: test/utils_test.dart [hunk 8]
```diff
@@ -0,0 +1,481 @@
...
+        true,
+      );
+      expect(
+        TooltipDefaultActionType.values.contains(TooltipDefaultActionType.skip),
+        true,
+      );
+    });
+  });
+
+  group('Model Edge Cases', () {
```
Comment (aditya-css) [inline test/utils_test.dart:119]: Seems unnecessary.

File: test/utils_test.dart [hunk 9]
```diff
@@ -0,0 +1,481 @@
...
+            ),
+          ),
+        ),
+      );
+
+      expect(find.byType(FloatingActionWidget), findsOneWidget);
+      expect(find.byIcon(Icons.star), findsOneWidget);
+    });
+
+    testWidgets('FloatingActionWidget renders correctly',
```
Comment (aditya-css) [inline test/utils_test.dart:164]: Repeated test.

File: test/utils_test.dart [hunk 10]
```diff
@@ -0,0 +1,481 @@
...
+        ),
+      );
+
+      expect(find.byType(FloatingActionWidget), findsOneWidget);
+      expect(find.byIcon(Icons.star), findsOneWidget);
+    });
+  });
+
+  group('Widget Edge Cases', () {
+    testWidgets('FloatingActionWidget with both top and bottom set',
```
Comment (aditya-css) [inline test/utils_test.dart:192]: Could be combined.

File: test/utils_test.dart [hunk 11]
```diff
@@ -0,0 +1,481 @@
...
+      () {
+        ShowcaseView.register();
+      },
+    );
+    tearDown(
+      () {
+        ShowcaseView.get().unregister();
+      },
+    );
+    testWidgets('Constants integration with Showcase',
```
Comment (aditya-css) [inline test/utils_test.dart:241]: Should also test that showcase's title or description is not found and showcase is not running.

File: test/utils_test.dart [hunk 12]
```diff
@@ -0,0 +1,481 @@
...
+              ),
+            ),
+          ),
+        ),
+      );
+
+      expect(find.text('Target Widget'), findsOneWidget);
+    });
+
+    testWidgets('Enum integration with Showcase', (WidgetTester tester) async {
```
Comment (aditya-css) [inline test/utils_test.dart:268]: Test doesn't match with its description.

File: test/utils_test.dart [hunk 13]
```diff
@@ -0,0 +1,481 @@
...
+              ),
+            ),
+          ),
+        ),
+      );
+
+      expect(find.text('Target Widget'), findsOneWidget);
+    });
+
+    testWidgets('Action buttons integration', (WidgetTester tester) async {
```
Comment (aditya-css) [inline test/utils_test.dart:293]: Test doesn't match with its description.

File: test/utils_test.dart [hunk 14]
```diff
@@ -0,0 +1,481 @@
...
+              ),
+            ),
+          ),
+        ),
+      );
+
+      expect(find.text('Target Widget'), findsOneWidget);
+    });
+
+    testWidgets('Floating action widget integration',
```
Comment (aditya-css) [inline test/utils_test.dart:329]: Already covered. Doesn't qualify as an integration test.

File: test/utils_test.dart [hunk 15]
```diff
@@ -0,0 +1,481 @@
...
+      ShowcaseView.get().startShowCase([key]);
+      await tester.pump();
+      await tester.pump(const Duration(milliseconds: 600));
+
+      expect(find.text('Floating Action'), findsOneWidget);
+    });
+  });
+
+  group('Error Handling Tests', () {
+    testWidgets('Invalid overlay opacity throws assertion error',
```
Comment (aditya-css) [inline test/utils_test.dart:369]: This test doesn't fall under utils.

File: test/utils_test.dart [hunk 16]
```diff
@@ -0,0 +1,481 @@
...
+            height: 50,
+            color: Colors.red,
+            child: const Text('Target Widget'),
+          ),
+        ),
+        throwsAssertionError,
+      );
+    });
+
+    testWidgets('Invalid targetTooltipGap throws assertion error',
```
Comment (aditya-css) [inline test/utils_test.dart:392]: This test doesn't fall under utils.

File: test/utils_test.dart [hunk 17]
```diff
@@ -0,0 +1,481 @@
...
+            color: Colors.red,
+            child: const Text('Target Widget'),
+          ),
+        ),
+        throwsAssertionError,
+      );
+    });
+
+    testWidgets(
+        'Invalid disposeOnTap and onTargetClick combination throws assertion error',
```
Comment (aditya-css) [inline test/utils_test.dart:416]: This test doesn't fall under utils.

File: test/utils_test.dart [hunk 18]
```diff
@@ -0,0 +1,481 @@
...
+            color: Colors.red,
+            child: const Text('Target Widget'),
+          ),
+        ),
+        throwsAssertionError,
+      );
+    });
+
+    testWidgets(
+        'Invalid onBarrierClick and disableBarrierInteraction combination throws assertion error',
```
Comment (aditya-css) [inline test/utils_test.dart:458]: This test doesn't fall under utils.
